src/ucp/core/ucp_worker.c

```diff
@@ -3031,7 +3029,7 @@ ucp_worker_do_keepalive_progress(ucp_worker_h worker)
         progress_count++;
         worker->keepalive.ep_count++;
     } while ((worker->keepalive.ep_count < max_ep_count) &&
-             (worker->keepalive.iter != worker->keepalive.iter_end));
+             (worker->keepalive.iter != &worker->all_eps));
```

Comment:
should we also change `if (worker->keepalive.iter == &worker->all_eps) {` check in ucp_worker_keepalive_next_ep?

Reply:
no, it shouldn't be changed.
if the iterator points to `all_eps`, the function will return `NULL` and then iteration will be stopped since `iter == &all_eps`

Reply:
ok